packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -0,0 +1,165 @@
+                    _logger.debug("Creating unshared GrpcChannelPool.")
+                    self._grpc_channel_pool = GrpcChannelPool()
+                if self._discovery_client is None:
+                    _logger.debug("Creating unshared DiscoveryClient.")
+                    self._discovery_client = DiscoveryClient(
+                        grpc_channel_pool=self._grpc_channel_pool
+                    )
+                if self._stub is None:
+                    service_location = self._discovery_client.resolve_service(
+                        provided_interface=_V2_MEASUREMENT_SERVICE_INTERFACE,
+                        service_class=self._service_class,
+                    )
+                    channel = self._grpc_channel_pool.get_channel(service_location.insecure_address)
+                    self._stub = v2_measurement_service_pb2_grpc.MeasurementServiceStub(channel)
+        return self._stub
+
+    def _create_file_descriptor(self) -> None:
+        metadata = self._get_stub().GetMetadata(v2_measurement_service_pb2.GetMetadataRequest())
+        input_metadata =  []        
+        for configuration in metadata.measurement_signature.configuration_parameters:
+            input_metadata.append(
+                ParameterMetadata.initialize(
+                    display_name=configuration.name,
+                    type=configuration.type,
+                    repeated=configuration.repeated,
+                    default_value=None,
+                    annotations=dict(configuration.annotations.items()),
+                    message_type=configuration.message_type,
+                )
+            )
+
+        output_metadata = []
+        for output in metadata.measurement_signature.outputs:
+            output_metadata.append(
+                ParameterMetadata.initialize(
+                    display_name=output.name,
+                    type=output.type,
+                    repeated=output.repeated,
+                    default_value=None,
+                    annotations=dict(output.annotations.items()),
+                    message_type=output.message_type,
+                )
+            )
+
+        create_file_descriptor(
+            input_metadata=input_metadata,
+            output_metadata=output_metadata,
+            service_name=self._service_class,
+            pool=descriptor_pool.Default(),
+        )
+
+    def measure(
+        self,
+        ${configuration_parameters_with_type_and_default_values}
+    ) -> ${output_type} :
+        """Executes ${display_name}.
+
+        Returns:
+            Measurement output.
+        """
+        parameter_values = [${measure_api_parameters}]
+        serialized_configuration = any_pb2.Any(
+            value=serialize_parameters(
+                parameter_metadata_dict=self._configuration_metadata, 
+                parameter_values=parameter_values, 
+                service_name=self._service_class +  ".Configurations"
+            )
+        )
```

Comment:
This does not work with C# services. 

Here's what I get when I try it:
```
PS D:\dev\measurement-plugin-python\packages\generator> poetry run ni-measurement-plugin-client-generator ni.examples.GameOfLife
PS D:\dev\measurement-plugin-python\packages\generator> poetry run python
Python 3.9.13 (tags/v3.9.13:6de2ca5, May 17 2022, 16:36:42) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import game_of_life_client
>>> client = game_of_life_client.GameOfLifeClient()
>>> client.measure()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\dev\measurement-plugin-python\packages\generator\game_of_life_client.py", line 192, in measure
    for response in self._get_stub().Measure(request):
  File "D:\dev\measurement-plugin-python\packages\generator\.venv\lib\site-packages\grpc\_channel.py", line 543, in __next__
    return self._next()
  File "D:\dev\measurement-plugin-python\packages\generator\.venv\lib\site-packages\grpc\_channel.py", line 969, in _next
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.UNKNOWN
        details = "Exception was thrown by handler."
        debug_error_string = "UNKNOWN:Error received from peer  {grpc_message:"Exception was thrown by handler.", grpc_status:2, created_time:"2024-08-28T18:45:14.0185954+00:00"}"
>
```
C:\ProgramData\National Instruments\Plug-Ins\Logs\DiscoveryService.txt:
```
===== 08/28/2024 13:45:09 ===== Information: ResolveService "ni.measurementlink.measurement.v2.MeasurementService": Local IP: "127.0.0.1" Remote IP: "127.0.0.1" DeploymentTarget: ""
===== 08/28/2024 13:45:09 ===== Information: Resolved "127.0.0.1" to HostName: "replicas"
===== 08/28/2024 13:45:09 ===== Information: HTTP "POST" "/ni.measurementlink.discovery.v1.DiscoveryService/ResolveService" responded 200 in 0.7390 ms
===== 08/28/2024 13:45:09 ===== Information: "Game Of Life (.NET)": "[13:45:09 INF] HTTP POST /ni.measurementlink.measurement.v2.MeasurementService/GetMetadata responded 200 in 0.8750 ms"
===== 08/28/2024 13:45:14 ===== Information: "Game Of Life (.NET)": "[13:45:14 ERR] Error when executing service method 'Measure'."
===== 08/28/2024 13:45:14 ===== Information: "Game Of Life (.NET)": "Google.Protobuf.InvalidProtocolBufferException: Full type name for GameOfLifeParameters is ni.measurementlink.examples.v1.GameOfLifeParameters; Any message's type url is "
===== 08/28/2024 13:45:14 ===== Information: "Game Of Life (.NET)": "   at Google.Protobuf.WellKnownTypes.Any.Unpack[T]()"
===== 08/28/2024 13:45:14 ===== Information: "Game Of Life (.NET)": "   at NationalInstruments.MeasurementLink.Examples.GameOfLifeMeasurement.MeasureAsync(MeasureRequest request, IServerStreamWriter`1 responseWriter, ServerCallContext context) in C:\dev\asw\Source\MeasurementServices\Examples\GameOfLife\Services\GameOfLifeMeasurement.cs:line 36"
===== 08/28/2024 13:45:14 ===== Information: "Game Of Life (.NET)": "   at NationalInstruments.MeasurementLink.Measurement.MeasurementServiceV2`2.Measure(MeasureRequest request, IServerStreamWriter`1 responseStream, ServerCallContext context) in C:\dev\asw\Source\MeasurementServices\MeasurementService\MeasurementServiceV2.cs:line 42"
===== 08/28/2024 13:45:14 ===== Information: "Game Of Life (.NET)": "   at Grpc.Shared.Server.ServerStreamingServerMethodInvoker`3.Invoke(HttpContext httpContext, ServerCallContext serverCallContext, TRequest request, IServerStreamWriter`1 streamWriter)"
===== 08/28/2024 13:45:14 ===== Information: "Game Of Life (.NET)": "   at Grpc.Shared.Server.ServerStreamingServerMethodInvoker`3.Invoke(HttpContext httpContext, ServerCallContext serverCallContext, TRequest request, IServerStreamWriter`1 streamWriter)"
===== 08/28/2024 13:45:14 ===== Information: "Game Of Life (.NET)": "   at Grpc.AspNetCore.Server.Internal.CallHandlers.ServerStreamingServerCallHandler`3.HandleCallAsyncCore(HttpContext httpContext, HttpContextServerCallContext serverCallContext)"
===== 08/28/2024 13:45:14 ===== Information: "Game Of Life (.NET)": "   at Grpc.AspNetCore.Server.Internal.CallHandlers.ServerCallHandlerBase`3.<HandleCallAsync>g__AwaitHandleCall|8_0(HttpContextServerCallContext serverCallContext, Method`2 method, Task handleCall)"
===== 08/28/2024 13:45:14 ===== Information: "Game Of Life (.NET)": "[13:45:14 INF] HTTP POST /ni.measurementlink.measurement.v2.MeasurementService/Measure responded 200 in 0.9208 ms"
```

Reply:
It fails because the `Any` message's `type_url` field is unset.
https://github.com/protocolbuffers/protobuf/blob/main/csharp/src/Google.Protobuf/WellKnownTypes/AnyPartial.cs#L65

See how the C# MeasurementClient calculates the type_url:
https://dev.azure.com/ni/DevCentral/_git/ASW?path=/Source/MeasurementServices/MeasurementClient/ParameterSerializer.cs&version=GBmain&line=79&lineEnd=80&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
Related: Python services are not validating the type_url.
https://github.com/ni/measurement-plugin-python/issues/850

Reply:
I filed a separate bug about this: https://github.com/ni/measurement-plugin-python/issues/851